src/main/java/frc/robot/autos/TwoRingSourceSide.java

```diff
@@ -22,21 +21,23 @@ public TwoRingSourceSide(SwerveDrive swerve, RotatorSubsystem rotator, ShooterSu
                         IntakeSubsystem intake) {
                 addCommands(
                                 new InstantCommand(() -> swerve.setInitialPose("subwoofer to middle ring 4 pickup")),
-                                new ReadyShooter(shooter, rotator, intake, swerve, map),
+                                new ReadyShooter(shooter, rotator, intake, 
+                                                () -> swerve.getDistanceMetersToGoal()),
                                 new RunCommand(() -> intake.setIntakeSpeed(0.8), intake).withTimeout(0.5),
                                 new InstantCommand(() -> intake.setIntakeSpeed(1)),
 
                                 new StowPreset(rotator, shooter),
                                 new WaitCommand(0.5),
                                 new ParallelCommandGroup(
-                                                swerve.followPathCommand("subwoofer to middle ring 4 pickup", false),
+                                                swerve.followPathCommand("subwoofer to middle ring 4 pickup"),
                                                 new InstantCommand(() -> intake.setIntakeSpeed(1))),
 
                                 new WaitCommand(0.5),
                                 new InstantCommand(() -> intake.setIntakeSpeed(0)),
                                 new ParallelCommandGroup(
-                                                swerve.followPathCommand("middle to shoot variation", false)),
-                                new ReadyShooter(shooter, rotator, intake, swerve, map),
+                                                swerve.followPathCommand("middle to shoot variation")),
+                                new ReadyShooter(shooter, rotator, intake, 
+                                                () -> swerve.getDistanceMetersToGoal()),
```

Comment:
yea, i think it's fine. we're really just utilizing swerve subsystem because that's where the method is held. it might get removed anyway, as that's a long path, so we really wanna do a FollowPathAndReadyShooter to cut down on timing. but yea, interesting to think about.